judge/question_navigator.py

```
@@ -0,0 +1,250 @@
+        self.question_flow_data, self.question_order = self._parse_rubric()
+
+    def _parse_rubric(self) -> tuple[Dict[str, Any], List[str]]:
+        """
+        Parse the rubric file into a navigable data structure.
+
+        The rubric has questions with potential multi-row answer options.
+        Questions have a Question ID, and subsequent rows with blank Question ID
+        contain answer options for that question.
+
+        Returns:
+            Tuple of (questions_dict, question_order_list):
+            - questions_dict: Dictionary mapping Question ID to question data
+            - question_order_list: Ordered list of Question IDs as they appear in the file
+        """
+        questions = {}
+        question_order = []
+        current_question_id = None
+        current_question_data = None
+
+        for idx, row in self.rubric_df.iterrows():
+            question_id_raw = row['Question ID'] if pd.notna(row['Question ID']) else None
+            # Convert to string and clean up (remove .0 from floats)
+            if question_id_raw is not None:
+                question_id = str(int(question_id_raw)) if isinstance(question_id_raw, (int, float)) else str(question_id_raw).strip()
+            else:
+                question_id = ""
+
+            # If this row has a Question ID, it's a new question
+            if question_id and question_id != 'nan':
+                # Save previous question if exists
+                # It means the current question is complete, so we need to save it
+                # we dont know if a question is complte until we get to a new one or to the end of the file
+                if current_question_id and current_question_data:
+                    questions[current_question_id] = current_question_data
+
+
+                # Read severity from the question row (not from answers)
+                # read the severity from the row, and check that is not empty, or just spaces
```

Are there legit cases where the Severity would be empty?  If not, I feel like we should throw an error here, rather than make all the allowances for it being possibly empty.

---

I'm also not sure why this needs separate handling from the rest of the row down below in line 71-78.